contracts/strategies/idle-tranche/IdleFinanceTrancheStrategy.sol

```diff
@@ -0,0 +1,320 @@
+pragma solidity 0.5.16;
+
+import "@openzeppelin/contracts/math/Math.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+
+import "../../base/interface/IStrategy.sol";
+import "../../base/upgradability/BaseUpgradeableStrategyUL.sol";
+
+import "./interface/IIdleCDO.sol";
+import "./interface/IWstETH.sol";
+import "./interface/IDistributor.sol";
+import "./interface/ILiquidityGaugeV3.sol";
+
+contract IdleFinanceTrancheStrategy is IStrategy, BaseUpgradeableStrategyUL {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    // additional storage slots (on top of BaseUpgradeableStrategy ones) are defined here
+    bytes32 internal constant _IS_STETH = 0x9574989eae0060a39fe95049948b9a7bc2cfe07156ccbe6fea8f4f977bebcc04;
+    bytes32 internal constant _IDLE_DISTRIBUTOR = 0xb2f7d491f85e6b66612eb5a10a2ed5d8285b176fec369924ce35cc46190bb048;
+    bytes32 internal constant _LIQUIDITY_GAUGE = 0xdefb28cb4afe76becb36959ca990fb40075f668b97d0dbfc4c4da2c6e73e879a;
+    bytes32 internal constant _LIQUIDITY_GAUGE_REWARD = 0xdb1165a9ca87102cffa27e10807f703b726c6ddabff7e9377e7b206ce2945fea;
+
+    address public constant WETH = address(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+    address public constant STETH = address(0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84);
+    address public constant WSTETH = address(0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0);
+    address public constant IDLE = address(0x875773784Af8135eA0ef43b5a374AaD105c5D39e);
+
+    uint256 public constant ONE_TRANCHE_TOKEN = 10**18;
+
+    constructor() public BaseUpgradeableStrategyUL() {
+        assert(_IS_STETH == bytes32(uint256(keccak256("eip1967.strategyStorage.isstETH")) - 1));
+        assert(_IDLE_DISTRIBUTOR == bytes32(uint256(keccak256("eip1967.strategyStorage.idleDistributor")) - 1));
+        assert(_LIQUIDITY_GAUGE == bytes32(uint256(keccak256("eip1967.strategyStorage.liquidityGauge")) - 1));
+        assert(_LIQUIDITY_GAUGE_REWARD == bytes32(uint256(keccak256("eip1967.strategyStorage.liquidityGaugeReward")) - 1));
+    }
+
+    /* ========== Initialize ========== */
+
+    function initializeBaseStrategy(
+        //  "__" for storage because we shadow _storage from GovernableInit
+        address __storage,
+        address _underlying,
+        address _vault,
+        address _rewardPool,
+        address _idleDistributor,
+        address _liquidityGauge,
+        address _liquidityGaugeReward
+    ) public initializer {
+        BaseUpgradeableStrategyUL.initialize({
+            _storage: __storage,
+            _underlying: _underlying,
+            _vault: _vault,
+            _rewardPool: _rewardPool,
+            _rewardToken: WETH,
+            _profitSharingNumerator: 300,
+            _profitSharingDenominator: 1000,
+            _sell: true,
+            _sellFloor: 1e18,
```

Comment:
This sellFloor is too high with WETH as rewardToken. It will now only liquidate if the total rewards are more than 1ETH. sellFloor could actually be set to 0, as we simulate profitability before running hardWork anyways.